src/main/scala/tech/cryptonomic/conseil/tezos/TezosRemotes.scala

```diff
@@ -0,0 +1,202 @@
+      import cats.syntax.apply._
+      import scala.concurrent.ExecutionContext
+      import scala.util.control.NoStackTrace
+
+      //might be actually unlawful, but we won't use it for append, only for empty
+      private implicit def futureMonoid[T](implicit ec: ExecutionContext) = new Monoid[Future[T]] {
+        override def combine(x: Future[T], y: Future[T]) = x *> y
+        override def empty: Future[T] = Future.failed(new IllegalStateException("Tezos node requests will no longer be accepted.") with NoStackTrace)
+      }
+
+      type JustString[CallId] = Const[String, CallId]
+
+      implicit def futuresInstance(implicit context: RemoteContext) =
+        new RemoteRpc[Future, Id, JustString] {
+          import context._
+
+          private val logger = Logger("Akka.Futures.RemoteRpc")
+
+          implicit val system = context.system
+          implicit val dispatcher = system.dispatcher
+          implicit val materializer = ActorMaterializer()
+
+          type CallConfig = Any
+          type PostPayload = JsonString
+
+          def runGetCall[CallId](
+            callConfig: Any = (),
+            request: CallId,
+            commandMap: CallId => String
+          ): Future[JustString[CallId]] = withRejectionControl {
+
+            val url = (commandMap andThen translateCommandToUrl)(request)
+            val httpRequest = HttpRequest(HttpMethods.GET, url)
+            logger.debug("Async querying URL {} for platform Tezos and network {}", url, tezosConfig.network)
+
+            for {
+              response <- Http(system).singleRequest(httpRequest)
+              strict <- response.entity.toStrict(requestConfig.GETResponseEntityTimeout)
+            } yield Const(JsonString sanitize strict.data.utf8String)
+
+          }
+
+          def runPostCall[CallId](
+            callConfig: Any = (),
+            request: CallId,
+            commandMap: CallId => String,
+            payload: Option[JsonString]
+          ): Future[JustString[CallId]] = withRejectionControl {
+
+            val url = (commandMap andThen translateCommandToUrl)(request)
+            logger.debug("Async querying URL {} for platform Tezos and network {} with payload {}", url, tezosConfig.network, payload)
+            val postedData = payload.getOrElse(JsonString.emptyObject)
+            val httpRequest = HttpRequest(
+              HttpMethods.POST,
+              url,
+              entity = HttpEntity(ContentTypes.`application/json`, postedData.json.getBytes())
+            )
+
+            for {
+              response <- Http(system).singleRequest(httpRequest)
+              strict <- response.entity.toStrict(requestConfig.POSTResponseEntityTimeout)
+            } yield {
+              val responseBody = strict.data.utf8String
+              logger.debug("Query results: {}", responseBody)
+              Const(JsonString sanitize responseBody)
+            }
+          }
+        }
+    }
+
+    object Streams {
```

Comment:
This does not look like a typical companion object. Does it make more sense to give it a slightly different name?

Reply:
Maybe we want to define some rule here about naming.

I am accustomed to use `object` in scala as a sort of locally-defined piece of modularity, that is, a _module_.

In this case, the name choice is driven by its expected usage in locally-scoped `import` directives.

It's much like cats defines (package) objects like `cats.instances.yourclassname._`, `cats.syntax.typeclassname._` and so on.

These objects are not meant to be called for the behaviour provided, but to be imported for the instances they bring in scope, they're more akin to namespaces than singleton static instances.

This was a design goal of the language, along with mixin-composition, to provide modular extensible behaviour, as outlined in [this post](http://lambda-the-ultimate.org/node/3735), whose last reference should point to [this paper by Odersky & Zenger](https://lampwww.epfl.ch/~odersky/papers/ScalableComponent.pdf)